### PR TITLE
core: add quotes to the json keys produced by function calling

### PR DESCRIPTION
**Pull request by nadworny**
- **Description:** Current Azure OpenAI models provide invalid json (keys don't contain quotes) which breaks the llm.with_structured_output functionality.
- **Issue:** #19699 
- **Dependencies:** None
- **Twitter handle:** None

### CODE REVIEW

**Comment by nadworny** [comment]
mentioning as suggested in the PR template @baskaryan

**Review by baskaryan — COMMENTED** [review]
could we add some unit tests

**Review by hwchase17 — COMMENTED** [review]
definitely needs unit tests

File: libs/core/langchain_core/output_parsers/openai_tools.py [hunk 1]
```diff
@@ -47,6 +48,18 @@ def parse_tool_call(
...
     return parsed
 
+def make_json_valid(json_string: str) -> str:
+    """
+    This function takes a dictionary-like string and converts it into a valid JSON string.
+    It does this by adding double quotes around the keys.
+    """
+    try:
+        corrected_json_string = re.sub(r'(\b\w+\b):', r'"\1":', json_string)
+        return corrected_json_string
```
Comment (rohan-patankar) [inline libs/core/langchain_core/output_parsers/openai_tools.py:59]: add corrected_json_string.replace("(", "").replace(")", "") add this to it bcz always getting in str(tuples)

File: libs/core/langchain_core/output_parsers/openai_tools.py [hunk 2]
```diff
@@ -47,6 +48,18 @@ def parse_tool_call(
         parsed["id"] = raw_tool_call.get("id")
     return parsed
 
+def make_json_valid(json_string: str) -> str:
+    """
+    This function takes a dictionary-like string and converts it into a valid JSON string.
+    It does this by adding double quotes around the keys.
+    """
+    try:
+        corrected_json_string = re.sub(r'(\b\w+\b):', r'"\1":', json_string)
```
Comment (rohanpatankar926) [inline libs/core/langchain_core/output_parsers/openai_tools.py]: add corrected_json_string.replace("(", "").replace(")", "") bcz always getting in str(tuple)
so this should fix the output format i have tried and working for me

**Comment by hwchase17** [comment]
closing due to inavticity

**Pull request closed without merging**